src/Idris/Parser.hs

```diff
@@ -1396,14 +1396,15 @@ pLangExt = (reserved "TypeProviders" >> return TypeProviders)
 {- | Parses a totality
 
 @
-Totality ::= 'partial' | 'total'
+Totality ::= 'partial' | 'total' | 'covering'
 @
 
 -}
 totality :: IdrisParser Bool
 totality
         = do reservedHL "total";   return True
       <|> do reservedHL "partial"; return False
+      <|> do reservedHL "covering"; return False
```

Comment:
So your issue is that the result should be changed from a Bool to somekind of Totality enum which is then propagated elsewhere